src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs

```diff
@@ -1653,5 +1644,24 @@ private static bool IsAzureDevOpsArtifactsUrl(string location)
             }
             return false;
         }
+
+        /// <summary>
+        /// Delete a file with retry. Sometimes when gathering a drop directly to a share
+        /// we can get occasional deletion failures. All of them seen so far are UnauthorizedAccessExceptions
+        /// </summary>
+        /// <param name="filePath">Full path to the file to delete.</param>
+        private static async Task DeleteFileWithRetryAsync(string filePath)
+        {
+            await ExponentialRetry.RetryAsync(
+                () =>
+                {
+                    if (File.Exists(filePath))
+                    {
+                        File.Delete(filePath);
+                    }
+                },
+                ex => Console.WriteLine($"Failed to delete {filePath}: {ex.Message}"),
+                ex => ex is UnauthorizedAccessException);
```

Comment:
We get `Unauthorized Access` and then it works on retry?

Reply:
Yep.